pkg/controller/container-runtime-config/helpers.go

```diff
@@ -209,39 +218,89 @@ func updateStorageConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfigura
 	return newData.Bytes(), nil
 }
 
-// updateCRIOConfig decodes the data rendered from the template, merges the changes in and encodes it
-// back into a TOML format. It returns the bytes of the encoded data
-func updateCRIOConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error) {
-	tomlConf := new(tomlConfigCRIO)
-	if _, err := toml.DecodeReader(bytes.NewBuffer(data), tomlConf); err != nil {
-		return nil, fmt.Errorf("error decoding crio config: %v", err)
-	}
+// createCRIOdropinConfigLogLevel decodes the data rendered from the template, merges the changes in and encodes it
+// back into a TOML format. It returns the bytes of the encoded data.
+// This is only for when the log-level field is changed. We have separated it out into 3 different toml structs
```

Comment:
This feels like a comment for definitions of the three structures; that’s the place to explain why the CRI-O types are not reused, and why we need three different ones.

Reply:
yup, fixed